src/Morse/MorseEncoding.ts

```diff
@@ -0,0 +1,81 @@
+// The encoding for morse is much less straightforward than the others. The dots
+// and dashes each use two bits of the integer starting with the least-
+// significant bit.
+//
+// Example:
+//
+// 'F' -> ..-.
+//
+// Which encodes as:
+//
+// 01 01 10 01
+//  .  .  -  .
+//
+// Since it's packed starting with the LSB, the resulting number is reversed:
+//
+// 01100101 -> 0x65
+//
+
+enum MorseEncoding {
+  // Letters
```

Comment:
I'd love to see a clearer way to write this.  I don't know if JS has any sense of constexpr that would let us define enums as the result of functions or if we'd have to have code to generate this, but it would be nice to see something like:

BraillePack(characters : array<char>) {
    int result = 0;
    foreach (char c : characters) doBitMath;
}

///////
LetterA = BraillePack('.', '-'),

Reply:
Heck, it might be worth having (or even exporting) that function to make it easier to generate these things when adding new character sets.

Reply:
I had generated these using a script: https://github.com/kfarnung/puzzle-lib/commit/e37fb5e8d0e938b5920ba0a10a1bcdc0294c717b

I don't believe there's a constexpr equivalent.  For the encodings where it's just a bitwise OR I went ahead and left the slight inefficiency in there, but it seemed a little too inefficient for this.  That said, I can look into using a runtime function call to generate these enum values.

Reply:
Took a stab at it, let me know what you think.